lib/battle/BattleInfo.cpp

```diff
@@ -274,7 +275,7 @@ BattleInfo * BattleInfo::setupBattle(const int3 & tile, const Terrain & terrain,
 
 				for(BattleHex blocked : obstPtr->getBlockedTiles())

```

Comment:
Am I right that this obidgen is implemented incorrectly because it relays that absolute obstacle ids are in range 0 - X and now they start after regular obstacles. So it will often give id of regular obstacle and we will use it to lookup in absoluteObstacles?

Reply:
I begin to understand. Looks like we have two ids here. One is internal ContentLoader assigned index which is in objects collection and one - an index in obstacles/absoluteObstacles arrays. This looks like some mess. And indexes in JSON looks like just string ids and can be replaced with o1, o2, o3, o345 literals for isntance. Or by animation name.